// nyrr-results-api/build/httpHandler.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const axios_1 = __importDefault(require("axios"));
class HttpHandler {
    constructor(token) {
        this.baseUrl = 'https://results.nyrr.org/api';
        this.token = token;
    }
    postWithNyrrToken(endpoint, data) {
        return __awaiter(this, void 0, void 0, function* () {
            const url = `${this.baseUrl}/${endpoint}`;
            const response = yield axios_1.default.post(url, data, {
                headers: {
                    'content-type': 'application/json;charset=UTF-8',
                    'token': this.token,
                }
            });
            return response;
        });
    }
    static getToken() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield axios_1.default.get('https://results.nyrr.org/GetSettings/rms-settings.rjs');
            const jsonResponse = response.data.replace('var settings = ', '');
            const data = JSON.parse(jsonResponse);
            return data.token;
        });
    }
}
exports.default = HttpHandler;
