// nyrr-results-api/build/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const clubStandings = __importStar(require("./clubStandings"));
const awards = __importStar(require("./awards"));
const events = __importStar(require("./events"));
exports.default = {
    getYears() {
        return __awaiter(this, void 0, void 0, function* () {
            return yield clubStandings.getYears();
        });
    },
    getDivisionsResults(year) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield clubStandings.getDivisionsResults(year);
        });
    },
    getDivisionResults(divisionCode, year) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield clubStandings.getDivisionResults(divisionCode, year);
        });
    },
    getTeams(year) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield clubStandings.getTeams(year);
        });
    },
    getTeamAwards(eventCode, teamCode, gender = null, minimumAge = null) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield awards.getTeamAwards(eventCode, teamCode, gender, minimumAge);
        });
    },
    getTeamAwardRunners(eventCode, teamCode, teamGender = null, teamMinimumAge = null) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield awards.getTeamAwardRunners(eventCode, teamCode, teamGender, teamMinimumAge);
        });
    },
    searchEvents(year = null, searchString = "", distance = null) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield events.search(year, searchString, distance);
        });
    },
};
